paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/TableWriteOperator.java

```diff
@@ -52,37 +55,54 @@ public TableWriteOperator(
         this.initialCommitUser = initialCommitUser;
     }
 
+    private boolean needState() {
+        if (table.coreOptions().writeOnly()) {
+            // Commit user for writers are used to avoid conflicts.
```

Comment:
But users may use the commit user to determine if a table's snapshot is generated by the other job. See : https://github.com/apache/paimon/pull/3474

So, I think you can add a config for users to decide whether they need the commitUser state (default true). And explain in the doc that in bounded stream scenarios, if this configuration is true, there will be side effects(when chaining source and write operator together, if some source parallelism have finished, checkpoint cannot be performed).

Reply:
This `commitUser` is only for writers. What you have mentioned is the `commitUser` for comitters.

Reply:
ok